presto-main/src/main/java/com/facebook/presto/operator/scalar/RegexpFunctions.java

```diff
@@ -51,264 +43,104 @@ private RegexpFunctions()
 
     @ScalarOperator(OperatorType.CAST)
     @SqlType(RegexpType.NAME)
-    public static Regex castToRegexp(@SqlType(StandardTypes.VARCHAR) Slice pattern)
+    public static RegexpGenericPattern castToRegexp(ConnectorSession session, @SqlType(StandardTypes.VARCHAR) Slice pattern)
     {
-        Regex regex;
+        String regexLibraryName = session.getSystemProperty(REGEX_LIBRARY, String.class);
+        RegexLibrary regexLibrary;
+        if (regexLibraryName.toUpperCase().equals("RE2J")) {
+            regexLibrary = RE2J;
+        }
+        else if (regexLibraryName.toUpperCase().equals("JONI")) {
+            regexLibrary = JONI;
+        }
+        else {
+            throw new PrestoException(INVALID_SESSION_PROPERTY, "The setting for '" + REGEX_LIBRARY + "' is not valid. It should be either RE2J or JONI.");
+        }
+
+        int dfaStatesLimit = session.getSystemProperty(RE2J_DFA_STATES_LIMIT, Integer.class);
```

Comment:
According to https://github.com/google/re2j, re2j doesn't have DFA support. I understand that it's on your roadmap to add DFA support. But I assumed you haven't. Correct me if I'm wrong.